File: Day335/BuyTwoChocoloates.go
func buyChoco(prices []int, money int) int {
    min1, min2 := prices[0], math.MaxInt
    for _,v := range prices[1:] {
        if v < min1 {
            min2, min1 = min1, v
        } else if min2 > v {
                min2 = v
        }
    }
    // fmt.Println(min1, min2)
    if (min1 + min2) <= money {
        return money - min1 - min2
    }
    return money
}

File: Day335/BuyTwoChocoloates.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        int min1 = prices[0], min2 = Integer.MAX_VALUE;
        for (int i=1; i<prices.length; i++) {
            if (prices[i] < min1) {
                min2= min1;
                min1 = prices[i];
            } else if (min2 > prices[i]) {
                    min2 = prices[i];
            }
        }
        // fmt.Println(min1, min2)
        if ((min1 + min2) <= money) {
            return money - min1 - min2;
        }
        return money;
    }
}